bin/out

```diff
@@ -54,19 +81,42 @@ if [[ "$KUBE_URL" =~ https.* ]]; then
 fi
 
 export KUBECTL
+
 DEPLOYMENT_LIST=$(jq -r .source.deployment_list< "$payload")
 SANITIZED_DEPLOYMENT_LIST=`echo $DEPLOYMENT_LIST | sed -e 's,\,,,g' | sed -e 's,\[,,g' | sed -e 's,\],,g' | sed -e 's,\",,g'`
 DEPLOYMENT_ARRAY=($SANITIZED_DEPLOYMENT_LIST)
 
 ROLLBACK=$(jq -r .params.rollback < "$payload")
+DEPLOY=$(jq -r .params.deploy < "$payload")
+
 if [[ "$ROLLBACK" == "true" ]]; then
     echo "********************** Method invoked: Rollback **********************"
-fi
-
-if [[ "$ROLLBACK" = "true" ]]; then
     for deployment_name in "${DEPLOYMENT_ARRAY[@]}"; do
         rollback $deployment_name "$DRY_RUN";
     done
+elif [[ "$DEPLOY" == "true" ]]; then
+    echo "********************** Method invoked: Deploy Jobs **********************"
+
+    DEPLOYMENT_PATH=$(jq -r .source.deployment_file < "$payload")
+    IMG_PREFIX=$(jq -r .source.image_prefix < "$payload")
+
+    if [[ "$DEPLOYMENT_PATH" == "null" ]]; then
+        DEPLOYMENT_PATH=$(jq -r .params.deployment_file < "$payload")
+    fi
+
+    if [[ "$IMG_PREFIX" == "null" ]]; then
+        IMG_PREFIX=$(jq -r .params.image_prefix < "$payload")
+    fi
+
+    IMG_FILE=$(jq -r .params.image_name < "$payload")
+    IMG=$(cat "$IMG_FILE")
+    TAG_FILE=$(jq -r .params.image_tag < "$payload")
+    TAG=$(cat "$TAG_FILE")
+    IMG="$IMG:$TAG"
+
+    for deployment_name in "${DEPLOYMENT_ARRAY[@]}"; do
```

Comment:
this is the main change here, right?

Reply:
DEPLOYMENT_PATH won't be read per deployment in the array right now from what i see, since it's read off source. An Idea would be to do it in a rule based manner in the future.